controllers/v1/observations.js

```diff
@@ -67,35 +67,32 @@ async function instanceObservationData(req, res) {
         };
         resolve(response);
       } else {
-          let submissionId = req.body.submissionId;
+        let submissionId = req.body.submissionId;
         bodyData = req.body;
         let dataReportIndexes = await commonCassandraFunc.checkReqInCassandra(bodyData);
   
        if (dataReportIndexes == undefined) {
           model.MyModel.findOneAsync({ qid: "instance_observation_query" }, { allow_filtering: true })
             .then(async function (result) {
-  
+
               let bodyParam = JSON.parse(result.query);
-  
+
               if (config.druid.observation_datasource_name) {
                 bodyParam.dataSource = config.druid.observation_datasource_name;
               }
-              
+
+              bodyParam.filter.fields[0].value = submissionId;
 
               //if filter is given
-              if (req.body.filter) {
-                if (req.body.filter.questionId && req.body.filter.questionId.length > 0) {
-                  let filter = { "type": "and", "fields": [{ "type": "selector", "dimension": "observationSubmissionId", "value": submissionId }, { "type": "in","dimension":"questionExternalId","values": req.body.filter.questionId }] };
-                  bodyParam.filter = filter;
-                }
-                else {
-                  bodyParam.filter.value = submissionId;
-                }
+              if (req.body.filter && req.body.filter.questionId && req.body.filter.questionId.length > 0 ) {
```

Comment:
@deepakr123 - Optional questions should not be removed from the report and only child dependent questions if not answered we should skip

Reply:
resolved